### PR TITLE
Add `dask-deltatable`

### PR DESCRIPTION
**Pull request by charlesbluca**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by charlesbluca** [comment]
Opened https://github.com/dask-contrib/dask-deltatable/pull/60 to make sure the metadata between the recipe and package are consistent

File: recipes/dask-deltatable/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - pip
+
+about:
+  summary: Dask + Delta Table
+  license: BSD-3-Clause
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - charlesbluca
```
Comment (jacobtomlinson) [inline recipes/dask-deltatable/meta.yaml:43]:
```suggestion
    - charlesbluca
    - jacobtomlinson
```

Can you add me here? Might also want to add @jrbourbeau and @fjetter.

**Review by fjetter — APPROVED** [review]

File: recipes/dask-deltatable/meta.yaml (hunk 1 above)
Comment (jrbourbeau) [reply recipes/dask-deltatable/meta.yaml:44]: Thanks @charlesbluca @jacobtomlinson. I'm happy to be added

**Review by jrbourbeau — APPROVED** [review]

File: recipes/dask-deltatable/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  home: https://github.com/dask-contrib/dask-deltatable/
+  summary: Dask + Delta Table
+  license: BSD-3-Clause
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - charlesbluca
+    - fjetter
+    - jacobtomlinson
```
Comment (jakirkham) [inline recipes/dask-deltatable/meta.yaml:46]: Based on this comment ( https://github.com/conda-forge/staged-recipes/pull/23636#discussion_r1290498377 )

```suggestion
    - jacobtomlinson
    - jrbourbeau
```

File: recipes/dask-deltatable/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+test:
+  imports:
+    - dask_deltatable
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/dask-contrib/dask-deltatable/
```
Comment (jakirkham) [inline recipes/dask-deltatable/meta.yaml:37]: Minor nit: Would drop the trailing `/`

```suggestion
  home: https://github.com/dask-contrib/dask-deltatable
```

**Comment by jakirkham** [comment]
Thanks Charles! 🙏

Added a few comments above

**Comment by jakirkham** [comment]
Thanks Charles for the PR and everyone for reviews! 🙏

Let's follow up on anything else in the feedstock

**Pull request merged**